abb_driver/rapid/ROS_stateServer.mod

```diff
@@ -103,35 +107,45 @@ LOCAL PROC send_status()
     ENDTEST
 
     ! Get E-stop status
-    !  OR DOutput(doError)=1 OR DOutput(doMotorOffState)=1 THEN
-    IF DOutput(doRobEmgStop)=1 THEN
+    IF DOutput(signalRobotEStop) = 1 THEN
         message.e_stopped := ROS_TRISTATE_ON;
     ELSE
         message.e_stopped := ROS_TRISTATE_OFF;
     ENDIF
 
     ! Get whether motors have power
-    IF DOutput(doMotorOnState)=1 THEN
+    IF DOutput(signalMotorOn)=1 THEN
         message.drives_powered := ROS_TRISTATE_TRUE;
     ELSE
         message.drives_powered := ROS_TRISTATE_FALSE;
     ENDIF
 
-    ! TODO: Get error code
+    ! Get error code
+    message.error_code := ERRNO;
 
-    ! TODO: in_error
+    ! Determine in_error
+    if (message.error_code >= 1) AND (message.error_code <= 90) THEN
+        message.in_error := ROS_TRISTATE_TRUE;
+    ELSE
+        message.in_error := ROS_TRISTATE_FALSE;
+    ENDIF
 
     ! TODO: in_motion
-
+    !IF (ROS_new_trajectory) THEN

```

Comment:
This logic is problematic on real hardware. It keep trajectories from being executed.